src/tasks/minions/pickpocketActivity.ts

```diff
@@ -62,10 +64,24 @@ export default class extends Task {
 			return;
 		}
 		const currentLevel = user.skillLevel(SkillsEnum.Thieving);
+		let rogueOutfitBoostActivated = false;
 
 		const loot = new Bank();
 		for (let i = 0; i < successfulQuantity; i++) {
-			loot.add(npc.table.roll());
+			const lootItems = npc.table.roll();
+
+			if (rollRogueOutfitDoubleLoot(user)) {
+				lootItems.forEach(item => {
+					if (item.item === itemID('Rocky')) {
+						loot.add(item.item, item.quantity);
+					} else {
+						loot.add(item.item, item.quantity * 2);
+						rogueOutfitBoostActivated = true;
```

Comment:
instead of doing `rogueOutfitBoostActivated = true;` for every item, just put this as the first line in the if statement.